pkg/registry/apps/statefulset/strategy.go

```diff
@@ -75,11 +88,7 @@ func (statefulSetStrategy) PrepareForCreate(ctx context.Context, obj runtime.Obj
 	statefulSet.Generation = 1
 
 	pod.DropDisabledTemplateFields(&statefulSet.Spec.Template, nil)
-	if !utilfeature.DefaultFeatureGate.Enabled(features.MaxUnavailableStatefulSet) {
-		if statefulSet.Spec.UpdateStrategy.RollingUpdate != nil {
-			statefulSet.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable = nil
-		}
-	}
+	dropStatefulSetDisabledFields(statefulSet, nil)
```

Comment:
belongs to the first API commit

Reply:
is this absolutely necessary to separate it into a previous commit ?

Reply:
asking, because i dont know if there is a clean way to do that without doing the whole thing manually

Reply:
I think it should be in the correct commit especially since API review may be only done there. It's like 2 minute job with `tig`, reset/checkout and adding particular lines.

Reply:
@tnozicka this is resolved now , please take a look